how-to/customize-workspace/client/src/logger-shapes.ts

```diff
@@ -0,0 +1,69 @@
+export interface Logger<T = unknown> {
+	/**
+	 * Optionally initialize the logger.
+	 * @param options The custom options for the logger.
+	 */
+	initialize?(options: T): Promise<void>;
+
+	/**
+	 * Optionally close down the logger.
+	 */
+	closedown?(): Promise<void>;
+
+	/**
+	 * Log data as information.
+	 * @param group The group sending the log message.
+	 * @param message The message to log.
+	 * @param optionalParams Optional parameters for details.
+	 */
+	info(group: string, message: unknown, ...optionalParams: unknown[]): void;
+
+	/**
+	 * Log data as error.
+	 * @param group The group sending the log message.
+	 * @param message The message to log.
+	 * @param optionalParams Optional parameters for details.
+	 */
+	error(group: string, message: unknown, ...optionalParams: unknown[]): void;
+
+	/**
+	 * Log data as warning.
+	 * @param group The group sending the log message.
+	 * @param message The message to log.
+	 * @param optionalParams Optional parameters for details.
+	 */
+	warn(group: string, message: unknown, ...optionalParams: unknown[]): void;
+}
+
+export interface LoggerModule {
+	logger: Logger;
+}
+
+export interface LoggerModuleDefinition {
+	/**
+	 * The id of the module.
+	 */
+	id: string;
```

Comment:
should we have a base module definition that includes optional icon, description and info properties. This way if someone wants to build an admin panel they get some info about it.